add mark. add photo. rotate camera 90 degrees. take photo. ok with photo (return from camera). crash

### App information
* ID: de.dennisguse.opentracks
* Version: 4951 v4.0.3

### Device information
* Brand: samsung
* Device: a12s
* Model: SM-A127F
* Id: RP1A.200720.012
* Product: a12snseea

### Firmware
* SDK: 30
* Release: 11
* Incremental: A127FXXS5AVE1

### Cause of error
```java
Exception in thread "main": java.lang.RuntimeException: Unable to resume activity {de.dennisguse.opentracks/de.dennisguse.opentracks.ui.markers.MarkerEditActivity}: java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=963025608, result=-1, data=null} to activity {de.dennisguse.opentracks/de.dennisguse.opentracks.ui.markers.MarkerEditActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String android.net.Uri.toString()' on a null object reference
	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4918)
	at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:4955)
	at android.app.servertransaction.ResumeActivityItem.execute(ResumeActivityItem.java:52)
	at android.app.servertransaction.TransactionExecutor.executeLifecycleState(TransactionExecutor.java:176)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:97)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2336)
	at android.os.Handler.dispatchMessage(Handler.java:106)
	at android.os.Looper.loop(Looper.java:246)
	at android.app.ActivityThread.main(ActivityThread.java:8653)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:602)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1130)
Caused by: java.lang.RuntimeException: Failure delivering result ResultInfo{who=null, request=963025608, result=-1, data=null} to activity {de.dennisguse.opentracks/de.dennisguse.opentracks.ui.markers.MarkerEditActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String android.net.Uri.toString()' on a null object reference
	at android.app.ActivityThread.deliverResults(ActivityThread.java:5590)
	at android.app.ActivityThread.performResumeActivity(ActivityThread.java:4905)
	... 11 more
Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String android.net.Uri.toString()' on a null object reference
	at de.dennisguse.opentracks.ui.markers.MarkerEditViewModel.onNewCameraPhoto(MarkerEditViewModel.java:115)
	at de.dennisguse.opentracks.ui.markers.MarkerEditActivity.lambda$onCreate$4$de-dennisguse-opentracks-ui-markers-MarkerEditActivity(MarkerEditActivity.java:134)
	at de.dennisguse.opentracks.ui.markers.MarkerEditActivity$$ExternalSyntheticLambda3.onActivityResult(Unknown Source:4)
	at androidx.activity.result.ActivityResultRegistry.doDispatch(ActivityResultRegistry.java:392)
	at androidx.activity.result.ActivityResultRegistry.dispatchResult(ActivityResultRegistry.java:351)
	at androidx.activity.ComponentActivity.onActivityResult(ComponentActivity.java:647)
	at androidx.fragment.app.FragmentActivity.onActivityResult(FragmentActivity.java:164)
	at android.app.Activity.dispatchActivityResult(Activity.java:8550)
	at android.app.ActivityThread.deliverResults(ActivityThread.java:5583)
	... 12 more
```

--- a/src/main/java/de/dennisguse/opentracks/ui/markers/MarkerEditActivity.java
+++ b/src/main/java/de/dennisguse/opentracks/ui/markers/MarkerEditActivity.java
@@ -57,6 +57,8 @@ public class MarkerEditActivity extends AbstractActivity {
     public static final String EXTRA_TRACK_ID = "track_id";
     public static final String EXTRA_MARKER_ID = "marker_id";
 
+    private static final String CAMERA_PHOTO_URI_KEY = "camera_photo_uri_key";
+
     private static final String TAG = MarkerEditActivity.class.getSimpleName();
     private Track.Id trackId;
     private Marker marker;
@@ -87,6 +89,10 @@ public class MarkerEditActivity extends AbstractActivity {
 
         trackId = getIntent().getParcelableExtra(EXTRA_TRACK_ID);
         Marker.Id markerId = getIntent().getParcelableExtra(EXTRA_MARKER_ID);
+
+        if (savedInstanceState != null) {
+            cameraPhotoUri = Uri.parse(savedInstanceState.getString(CAMERA_PHOTO_URI_KEY, ""));
+        }
 
         hasCamera = getPackageManager().hasSystemFeature(PackageManager.FEATURE_CAMERA_ANY);
 
@@ -165,6 +171,14 @@ public class MarkerEditActivity extends AbstractActivity {
         viewModel = null;
         takePictureFromGallery = null;
         takePictureFromCamera = null;
+    }
+
+    @Override
+    protected void onSaveInstanceState(@NonNull Bundle outState) {
+        super.onSaveInstanceState(outState);
+        if (cameraPhotoUri != null) {
+            outState.putString(CAMERA_PHOTO_URI_KEY, cameraPhotoUri.toString());
+        }
     }
 
     @Override

